File: python-bindings/examples/basic.py
"""
Basic example of sending an email with micromail
"""

import micromail

def main():
    # Create a configuration
    config = micromail.Config("example.com")
    
    # Create a mailer
    mailer = micromail.Mailer(config)
    
    # Create an email
    mail = micromail.Mail()
    mail.from_addr("sender@example.com")
    mail.to_addr("recipient@example.com")
    mail.subject("Hello from Python")
    mail.body("This is a test email sent from Python!")
    
    # Send the email
    try:
        mailer.send(mail)
        print("Email sent successfully!")
        
        # Print the log
        print("\nLog:")
        for line in mailer.get_log():
            print(line)
    except RuntimeError as e:
        print(f"Failed to send email: {e}")

if __name__ == "__main__":
    main()

File: examples/basic.rs
//! Basic example of sending an email

use micromail::{Config, Mailer, Mail, Error};

fn main() -> Result<(), Error> {
    // Create a configuration
    let config = Config::new("example.com");
    
    // Create a mailer
    let mut mailer = Mailer::new(config);
    
    // Create an email
    let mail = Mail::new()
        .from("sender@example.com")
        .to("recipient@example.com")
        .subject("Hello from micromail")
        .body("This is a test email sent with micromail!");
    
    // Send the email
    match mailer.send_sync(mail) {
        Ok(_) => {
            println!("Email sent successfully!");
            
            // Print the log
            println!("\nLog:");
            for line in mailer.get_log() {
                println!("{}", line);
            }
        }
        Err(e) => {
            println!("Failed to send email: {}", e);
        }
    }
    
    Ok(())
}